test/unit/alignment/matrix/detail/two_dimensional_matrix_test.cpp

```diff
@@ -78,249 +167,183 @@ TYPED_TEST(two_dimensional_matrix_test, rows)
+    // cell content as accessing the original matrix by the same coordinate.
+    for (unsigned row = 0; row < this->matrix.rows(); ++row)
     {
-        using converted_t = two_dimensional_matrix<uint32_t, std::allocator<uint32_t>, matrix_major_order::row>;
-        converted_t converted = static_cast<converted_t>(matrix);
-        EXPECT_EQ(converted.rows(), matrix.rows());
-        EXPECT_EQ(converted.cols(), matrix.cols());
-        EXPECT_EQ((converted[{row_index_type{0u}, column_index_type{0u}}]), 0u);
-        EXPECT_EQ((converted[{row_index_type{1u}, column_index_type{0u}}]), 1u);
-        EXPECT_EQ((converted[{row_index_type{0u}, column_index_type{1u}}]), 3u);
-        EXPECT_EQ((converted[{row_index_type{0u}, column_index_type{3u}}]), 9u);
-        EXPECT_EQ((converted[{row_index_type{2u}, column_index_type{0u}}]), 2u);
-        EXPECT_EQ((converted[{row_index_type{2u}, column_index_type{3u}}]), 11u);
+        for (unsigned col = 0; col < this->matrix.cols(); ++col)
+        {
+            matrix_coordinate const idx{row_index_type{row}, column_index_type{col}};
+            new_score_type actual = converted_matrix[idx];
+            new_score_type expected = static_cast<new_score_type>(this->matrix[idx]);
+            this->expect_eq(actual, expected);
+        }
     }
 }
 
 //-----------------------------------------------------------------------------
 // Iterator tests
 //-----------------------------------------------------------------------------
 
-template <typename policy_t>
-struct iterator_fixture<two_dimensional_matrix_test<policy_t>> : two_dimensional_matrix_test<policy_t>
+template <typename score_t, matrix_major_order order>
+struct iterator_fixture<test_matrix_t<score_t, order>> : two_dimensional_matrix_test<test_matrix_t<score_t, order>>
 {
-    using base_t = two_dimensional_matrix_test<policy_t>;
-    using matrix_type = typename base_t::matrix_type;
+    using matrix_type = test_matrix_t<score_t, order>;
+    using base_t = two_dimensional_matrix_test<matrix_type>;
 
-    // Test random access range.
+    // test_range is a random access range.
     using iterator_tag = std::random_access_iterator_tag;
-    static constexpr bool const_iterable = true;
-
-    matrix_type test_range{number_rows{3}, number_cols{4}, base_t::expected_range};
+    static constexpr bool const_iterable = false;
+
+    // This one is a bit tricky. The one dimensional overloads of the iterator advancing operations, like it+=5, have
+    // the property that they advance the iterator in major matrix order, e.g. it+=5 in a row-wise matrix will be the
+    // same as it+=matrix_offset{row{0}, col{5}} and it+=5 in a column-wise matrix will be the same as
+    // it+=matrix_offset{row{5}, col{0}}. This effectively means that the iterator behaves exactly like the iterator of
+    // the internal data storage. Thus, passing expected_range as data storage into test_range, will result in an equal
```

Comment:
```suggestion
    // the internal data storage, i.e. the iterator's behaviour does not depend on the major matrix order. Thus, passing expected_range as data storage into test_range, will result in an equal
```

adjust line length if suggestion is added